scripts/case_selection/workflow_case_selection.R

```diff
@@ -76,8 +79,27 @@ prov_exec_selection <- "https://github.com/Sage-Bionetworks/genie-bpc-pipeline/t
 prov_exec_add <- prov_exec_selection
 prov_exec_report <- "https://github.com/Sage-Bionetworks/genie-bpc-pipeline/tree/develop/scripts/case_selection/perform_case_selection.Rmd"
 
+#' Get Main GENIE clinical file using release version name
+#'
+#' @param release Release version name for a GENIE consortium release
+#'                
+#' @return A named list of Main GENIE clinical file Synapse ID.
+get_main_genie_clinical_id <- function(release){
```

Comment:
Since it'll be quite the lift to have CI/CD for the tests and mocking in R, what do you recommend as a quick testing strategy?

Reply:
I was going to suggest filtering by version on the `data_clinical.txt` file directly but then I saw that the file's version comments are all `database`. 

Ahh the quickest I can think of is have a test (basically integration) that runs this function for a sample release say `17.1-consortium_release` and then check it against the actual synapse id it should output.

Reply:
This could be a common function. I didn't put it in there simply because I don't want to touch that file. lol. I can test the here package. If it fails, I will add it as a new ticket.

Reply:
> Ahh the quickest I can think of is have a test (basically integration) that runs this function for a sample release say 17.1-consortium_release and then check it against the actual synapse id it should output.

I tested it on my local when developing the function and it did output as expected.

Reply:
I think rixing was referring to using something like testthat R package and doing an integration test. 

Might be good practice if we are splitting off the function so the expectation is written out.

Reply:
In this particular scenario, I would be ok to be a bit lenient, but there are some upcoming work that we should absolutely add tests for as it pertains to mapping of values and business logic we need to ensure is right

Reply:
Got it. How about we discuss about adding test to code thoroughly next week? I found there are several test scripts both in tests folder and scripts folder, such as this: [test_dd_to_nonphi_dd.R](https://github.com/Sage-Bionetworks/genie-bpc-pipeline/blob/develop/scripts/dd/test_dd_to_nonphi_dd.R)

Reply:
Sounds good - one issue is that none of the tests are actually run on PR, so let's come up with a testing strategy that allows us to ensure high quality data delivery but also have confidence in the codebase and still allow us to meet our October delivery.

Reply:
Down to discuss overall testing strategy. Yeah I'm not sure if we can rely on those tests because they were updated a couple years ago. 

Right now I think we can only do ~two things for tests (just putting down some thoughts):
- Write tests (unit/integration) for new functions we add/functions we add new code to unless they are super long which then maybe we rely on the below method instead. Maybe like Tom suggested before, have something to just run those tests right before the rest of the code runs instead of integrating CI/CD
- Like Tom mentioned, integrate code to compare before and after (data quality testing)

I still think it would be good to add the test(s) (`testthat`) for that function in a separate test script before we merge the PR so we don't forget about it later with all of the BPC things coming up. Even if we can't automate the testing of it.

Reply:
I can add a test script under the case_selection folder then. Let's discuss an overall strategy next week.